tests/device_tests/mintlayer/test_get_public_key.py

```diff
@@ -0,0 +1,49 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2012-2019 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+import pytest
+
+from trezorlib import messages, mintlayer
+from trezorlib.debuglink import TrezorClientDebugLink as Client
+from trezorlib.tools import parse_path
+
+from ...input_flows import InputFlowShowXpubQRCode
+
+MINTLAYER_PATH = parse_path("m/44h/19788h/0h/0/0")
+
+
+@pytest.mark.altcoin
+@pytest.mark.mintlayer
+@pytest.mark.setup_client(
+    mnemonic="abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about"
+)
+def test_mintlayer_get_public_key(client: Client):
```

Comment:
> Though I'm still not sure how the network type should be passed to the firmware. The original code seems to use CoinInfo for that.

I think we should use `CoinInfo` too, so that we can call `validate_path_against_script_type` in `get_address/get_public_key/sign_message`, which will ensure that the derivation path is correct and show a warning to the user if it's not (as I can see from the code at least)